### PR TITLE
test: add tests for AssemblyScript SDK Transform

### PR DESCRIPTION
**Pull request by JairusSW**
**Implement tests for AssemblyScript SDK transform**

I implemented a basic test suite using node's builtin testing library which provided basic testing for the ASC Transform.

**Checklist**

- [x] Code compiles correctly and linting passes locally
- [x] For all _code_ changes, an entry added to the `CHANGELOG.md` file describing and linking to this PR
- [x] Tests added for new functionality, or regression tests for bug fixes added as applicable

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## UNRELEASED
 
+- feat: add tests for AssemblyScript SDK Transform [#659](https://github.com/hypermodeinc/modus/pull/659)
```
Comment (mattjohnsonpint) [inline CHANGELOG.md:5]:
```suggestion
- test: add tests for AssemblyScript SDK Transform [#659](https://github.com/hypermodeinc/modus/pull/659)
```

**Review by mattjohnsonpint — CHANGES_REQUESTED** [review]
Can you please write these in typescript?  Thanks.

**Comment by JairusSW** [comment]
@mattjohnsonpint yeah, I can do that
Just have to tweak the tsconfig a bit

**Comment by JairusSW** [comment]
@mattjohnsonpint I didn't end up compiling the tests to `js` with `tsc`, instead I added node's new `--experimental-strip-types` flag so it can execute the TypeScript files directly
Its experimental, so if you want, I can always switch it back to tsc, just lmk

**Comment by mattjohnsonpint** [comment]
For tests, the flag is fine as long as it works.  We can always replace it with `tsc` later if fails on anything.

Please add our standard copyright header to the newly added `.ts` files, and then I'll approve.  Thanks.

**Review by mattjohnsonpint — APPROVED** [review]

**Pull request merged**